src/solvers/quasi_Newton.jl

```diff
@@ -284,24 +289,24 @@ end
 
 function step_solver!(mp::AbstractManoptProblem, qns::QuasiNewtonState, iter)
     M = get_manifold(mp)
-    qns.X = get_gradient(mp, qns.p)
+    get_gradient!(mp, qns.X, qns.p)
     η = qns.direction_update(mp, qns)
     α = qns.stepsize(mp, qns, iter, η)
-    p_old = copy(M, get_iterate(qns))
+    copyto!(M, qns.p_old, get_iterate(qns))
     αη = α * η
     retract!(M, qns.p, qns.p, η, α, qns.retraction_method)
     β = locking_condition_scale(
-        M, qns.direction_update, p_old, αη, qns.p, qns.vector_transport_method
+        M, qns.direction_update, qns.p_old, αη, qns.p, qns.vector_transport_method
     )
     vector_transport_to!(
-        M, qns.sk, p_old, αη, qns.p, get_update_vector_transport(qns.direction_update)
+        M, qns.sk, qns.p_old, αη, qns.p, get_update_vector_transport(qns.direction_update)
     )
     vector_transport_to!(
-        M, qns.X, p_old, qns.X, qns.p, get_update_vector_transport(qns.direction_update)
+        M, qns.X, qns.p_old, qns.X, qns.p, get_update_vector_transport(qns.direction_update)
     )
-    new_grad = get_gradient(mp, qns.p)
-    qns.yk .= new_grad ./ β .- qns.X
-    update_hessian!(qns.direction_update, mp, qns, p_old, iter)
+    get_gradient!(mp, qns.new_grad, qns.p)
```

Comment:
yes, calling this `X_old` would require a copy here, I see; I would still prefer a unified naming.

Reply:
I've changed name to `X_old`.